### PR TITLE
fix: Desktop notification routing for direct rooms

### PR DESCRIPTION
**Pull request by yash-rajpal**
## Further comments

SUP-298

The notification payload might not have `name` property for direct rooms, so notifications might not redirect at right place.

### CODE REVIEW

**Comment by CvH** [comment]
does this fixes #30712 and #30733 introduced in 6.4 ?

**Comment by gabriellsh** [comment]
> does this fixes #30712 and #30733 introduced in 6.4 ?

Not related to this. I have some info on that so I'll comment there

File: apps/meteor/server/lib/rooms/roomCoordinator.ts [hunk 1]
```diff
@@ -40,13 +40,13 @@ class RoomCoordinatorServer extends RoomCoordinator {
 				sender: AtLeast<IUser, '_id' | 'name' | 'username'>,
 				notificationMessage: string,
 				userId: string,
-			): Promise<{ title: string | undefined; text: string }> {
+			): Promise<{ title: string | undefined; text: string; name: string | undefined }> {
 				const title = `#${await this.roomName(room, userId)}`;
-				const name = settings.get<boolean>('UI_Use_Real_Name') ? sender.name : sender.username;
+				const senderName = settings.get<boolean>('UI_Use_Real_Name') ? sender.name : sender.username;
```
[the remark this answers is not in the document]
Comment (yash-rajpal) [reply apps/meteor/server/lib/rooms/roomCoordinator.ts]: Nice catch!

**Review by gabriellsh — APPROVED** [review]
LGTM!

File: apps/meteor/server/lib/rooms/roomTypes/direct.ts [hunk 2]
```diff
@@ -94,16 +94,20 @@ roomCoordinator.add(DirectMessageRoomType, {
 	async getNotificationDetails(room, sender, notificationMessage, userId) {
 		const useRealName = settings.get<boolean>('UI_Use_Real_Name');
 
+		const roomName = room.name || (await this.roomName(room, userId));
+
 		if (this.isGroupChat(room)) {
 			return {
-				title: await this.roomName(room, userId),
+				title: roomName,
```
Comment (matheusbsilva137) [inline apps/meteor/server/lib/rooms/roomTypes/direct.ts:101]: The `roomName` function currently returns the `fname` (and not the `name`) property when `useRealName` is `true`. Shouldn't we keep this behavior?
If so, then mabe we can remove the changes from this line :thinking:

**Review by matheusbsilva137 — CHANGES_REQUESTED** [review]

File: apps/meteor/server/lib/rooms/roomTypes/direct.ts (hunk 2 above)
Comment (yash-rajpal) [reply apps/meteor/server/lib/rooms/roomTypes/direct.ts]: Done!

**Review by matheusbsilva137 — APPROVED** [review]
Back-end LGTM :+1:

**Review by tassoevan — APPROVED** [review]

**Pull request merged**